### PR TITLE
Remove mentions of semver for plugin-api

### PR DESCRIPTION
**Pull request by mdellweg**
[noissue]

### CODE REVIEW

File: CHANGES/7555.doc [hunk 1]
```diff
@@ -0,0 +1 @@
+Removed mentions of semver in the plugin API docs, and replaced them with a link to the deprecation policy where appropriate.
```
Comment (bmbouter) [inline CHANGES/7555.doc:1]: I see the main audience here as plugin writers so maybe rename this to `CHANGES/plugin_api/7555.doc` ?

Comment (mdellweg) [reply CHANGES/7555.doc:1]: Right. I keep forgetting about that.

File: docs/plugins/api-reference/index.rst [hunk 2]
```diff
@@ -1,19 +1,13 @@
 Plugin API Reference
 --------------------
 
-The Plugin API is versioned separately from the Pulp Core and consists of everything importable
-within the :mod:`pulpcore.plugin` namespace. When writing plugins, care should be taken to only
-import Pulp Core components exposed in this namespace; importing from elsewhere within the Pulp
-Core (e.g. importing directly from ``pulpcore.app``, ``pulpcore.exceptions``, etc.) is unsupported,
-and not protected by the Pulp Plugin API's semantic versioning guarantees.
+The Plugin API is versioned in sync with Pulp Core and consists of everything importable within the
```
Comment (bmbouter) [inline docs/plugins/api-reference/index.rst:4]: For clarity, I think we mean the package here, so I propose:

```suggestion
The Plugin API is versioned in sync with ``pulpcore`` and consists of everything importable within the
```

File: docs/plugins/api-reference/index.rst [hunk 3]
```diff
@@ -1,19 +1,13 @@
...
+:mod:`pulpcore.plugin` namespace. It is governed by our `deprecation policy <_deprecation_policy>`_.
+When writing plugins, care should be taken to only import Pulp Core components exposed in this
```
Comment (bmbouter) [inline docs/plugins/api-reference/index.rst:6]: same here 'Pulp Core'

File: docs/plugins/index.rst [hunk 4]
```diff
@@ -18,9 +18,8 @@ Let us know when the plugin is ready and we will be happy to add it to the list
 Plugin API
 ==========
 
-The Pulp Plugin API is versioned separately from Pulp Core. It is governed by `semantic
-versioning <http://semver.org/>`_. Backwards incompatible changes may be made until the
-Plugin API reaches stability with v1.0.
+The Pulp Plugin API is published and versioned with Pulp Core. It is governed by our `deprecation
```
Comment (bmbouter) [inline docs/plugins/index.rst:21]: similar here

**Review by bmbouter — APPROVED** [review]
This looks great, thank you!

**Pull request merged**